### ISSUE TITLE
Multiple file uploading via drag and drop to Firefox does not work

### ISSUE THREAD

**Issue by antalm** [doublecmd/doublecmd#1842]
### Version

Double Commander
Version: 1.1.17 gamma
Revision: 548
Commit: f4189f213
Build date: 2024/08/19
Lazarus: 3.4.0.0
FPC: 3.2.2
Platform: x86_64-Linux-qt5
OS version: Arch Linux

Widgetset library: Qt 5.15.15, libQt5Pas 5.6.1

### Description

If I select multiple files with insert key and then start dragging the selected files to a Firefox browser window (Firefox 130.0 (64-bit)) to a website which accepts drag and drop files (eg. https://wetransfer.com/), then only the file under the cursor is uploaded, not all of the selected files.
However if I drag and drop from Double Commander to Chrome browser, then it uploads all selected files just as expected.

First I thought this was simply a Firefox issue, but then I tried to upload multiple files from Thunar file manager to Firefox and it worked just fine.

So the issue seems to be some interworking problem between Double Commander 1.1.17 gamma and Firefox 130.0 (64-bit). Multiple file drag and drop used to work with possibly an earlier version of Double Commander and/or Firefox until about the month ago, but I am not sure what version I had of those previously.

**Comment by alexx2000** [comment]
I noticed this too. I guess something changed in Firefox because nothing changed in DC.

**Comment by Knusper** [comment]
This also happens with Dolphin and Firefox - since I haven't updated Dolphin in a while, it must have been caused by Firefox update.  I use the official Mozilla deb file of firefox on my system.

**Comment by alexx2000** [comment]
Then I think it's a Firefox bug, maybe related with https://bugzilla.mozilla.org/show_bug.cgi?id=1914742

**Comment by antalm** [comment]
Yes, it most probably is a Firefox bug, but why does multiple files drag and drop work from Thunar to Firefox though? I mean if this was a Firefox only bug then it should not work regardless of where the files are dragged from.
What I can think of is something like Firefox dropped the support of some legacy file drag method and eg. Thunar already uses a new one so it works fine, but Double Commander still uses a legacy API that Firefox no longer supports. But this is just a guess, I am absolutely not familiar with GUI APIs.

**Comment by kskonovalov** [comment]
Same issue here from drag&drop from Krusader. Here's what I noticed:

Krusader -> multiple files FF -> not working
Krusader -> multiple files Chrome -> works
Gnome Files -> multiple files FF -> works

**Comment by antalm** [comment]
Hi,

It seems to work now. I don't know exactly since when, because I was already half accustomed to having to drag & drop files from Double Commander to Firefox one by one, but I noticed that it works at least a few days ago with Double Commander 1.1.22 rev662 and Firefox 136.0.1.
